python/loyalty_workflow.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.common import WorkflowIDReusePolicy
+from temporalio.exceptions import WorkflowAlreadyStartedError
+from temporalio.workflow import ParentClosePolicy
+from temporalio.client import WorkflowHandle
+
+import email_strings
+from shared import *
+
+
+# Basic activity that logs and does string concatenation
+@activity.defn(name="SendEmail")
+async def send_email(body: str):
+    activity.logger.info("Sending email with contents %s" % body)
+
+
+# Basic workflow that logs and invokes an activity
+@workflow.defn
+class CustomerLoyaltyWorkflow:
+    def __init__(self):
+        self.customer = None
+
+    @workflow.run
+    async def run(self, customer: Customer) -> str:
+        self.customer = customer
+        workflow.logger.info("Running workflow with parameter %s" % customer)
+        info = workflow.info()
+
+        if not info.continued_run_id:
```

Comment:
Many of my comments on the Go workflow apply here about doing things like using workflow input to drive this decision rather than whether continued. It's often clearer to readers.

Reply:
I believe I carried those recommendations over, but as always a second look is very much appreciated!